tiny_dnn/nodes.h

```diff
@@ -43,21 +43,25 @@ namespace cereal {
 
 template <typename Archive>
 void save(Archive & ar, const std::vector<tiny_dnn::layerptr_t>& v) {
+#ifndef CNN_NO_SERIALIZATION
     ar(cereal::make_size_tag((cereal::size_type)v.size()));
```

Comment:
Throwing an exception or `assert(false);` might be good in case `#ifdef CNN_NO_SERIALIZATION`